### ISSUE TITLE
javacc/java: missing package declaration in generated Token class

### ISSUE THREAD

**Issue by zosrothko** [javacc/codegenerator#36]
Hi

Withe the grammar below, javacc-8.0.0 generate a Token with a missing package declaration
```
PARSER_BEGIN(Example)
package example1;

public class Example {

  public static void main(String args[]) throws ParseException {
    Example parser = new Example(System.in);
    parser.Input();
  }

}

PARSER_END(Example)

SKIP :
{
  " "
| "\t"
| "\n"
| "\r"
}

void Input() :
{}
{
  "a" BC() "c"
}

void BC() :
{}
{
  "b" [ "c" ]
}
```
Generated Token.java below with missing `package example1;`
```
/* Generated By:JavaCC: Do not edit this line. Token.java Version 8.0 */
/* JavaCCOptions:TOKEN_EXTENDS=,KEEP_LINE_COLUMN=true,SUPPORT_CLASS_VISIBILITY_PUBLIC=true */
/**
 * Describes the input token stream.
 */

public class Token {
```

**Comment by mbrigl** [comment]
With latest changes in #38  not anymore reproducible.

**Comment by zosrothko** [comment]
Still present on my side
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project lookahead: Compilation failure: Compilation failure:
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example10/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example6/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example2/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example9/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example1/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example8/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example7/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example5/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example4/Token.java:[7,8] duplicate class: Token
[ERROR] /C:/Users/fandre/Documents/git/javacc/javacc-8.0/examples/java/Lookahead/target/generated-sources/javacc/example8/ParseException.java:[33,25] cannot access example8.Token
[ERROR]   bad source file: C:\Users\fandre\Documents\git\javacc\javacc-8.0\examples\java\Lookahead\target\generated-sources\javacc\example8\Token.java
[ERROR]     file does not contain class example8.Token
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the sourcepath.
[ERROR] -> [Help 1]
```
```
/* Generated By:JavaCC: Do not edit this line. Token.java Version 8.0 */
/* JavaCCOptions:TOKEN_EXTENDS=,KEEP_LINE_COLUMN=true,SUPPORT_CLASS_VISIBILITY_PUBLIC=true */
/**
 * Describes the input token stream.
 */

public class Token {

  public int kind;
```

**Comment by mbrigl** [comment]
Still not able to reproduce it. Following main generates the Token class below

```
public class JavaCCMain {

  public static final String BASE = System.getProperty("user.dir") + "/src/main/java/";

  public static void main(String[] args) throws Exception {
    String[] arguments = new String[] { 
        "-CODE_GENERATOR=Java", 
        "-OUTPUT_DIRECTORY=" + BASE + "example1",
        BASE + "example1" + "/test.jjt" };

    JJTree jjtree = new JJTree();
    jjtree.main(arguments);

    arguments[2] = BASE + "example1" + "/test.jj";

    Main.mainProgram(arguments);
  }
}
```

```
/* Generated By:JavaCC: Do not edit this line. Token.java Version 8.0 */
/* JavaCCOptions:TOKEN_EXTENDS=,KEEP_LINE_COLUMN=true,SUPPORT_CLASS_VISIBILITY_PUBLIC=true */
package example1;

/**
 * Describes the input token stream.
 */

public class Token {

  public int kind;
```

**Comment by zosrothko** [comment]
Unless I misunderstand, the JavaCCMain is invoking JJTree then JavaCC while the grammar in concern is Example1.jj, a JavaCC grammar

**Comment by zosrothko** [comment]
With the latest 8.0.0 branch, the package statement is still missing in the generated Token class
```
Z:\git\javacc\javacc\javacc>rd /q /s tmp

Z:\git\javacc\javacc\javacc>javacc.bat -OUTPUT_DIRECTORY=tmp examples\java\Lookahead\src\main\javacc\Example1.jj

Z:\git\javacc\javacc\javacc>setlocal

Z:\git\javacc\javacc\javacc>set M2=C:\Users\FrancisANDRE\.m2\repository\org\javacc

Z:\git\javacc\javacc\javacc>set PARSER=C:\Users\FrancisANDRE\.m2\repository\org\javacc\parser\8.0.0\parser-8.0.0.jar

Z:\git\javacc\javacc\javacc>set CPP=C:\Users\FrancisANDRE\.m2\repository\org\javacc\codegen\cpp\8.0.0\cpp-8.0.0.jar

Z:\git\javacc\javacc\javacc>set JAVA=C:\Users\FrancisANDRE\.m2\repository\org\javacc\codegen\java\8.0.0\java-8.0.0.jar

Z:\git\javacc\javacc\javacc>set CSHARP=C:\Users\FrancisANDRE\.m2\repository\org\javacc\codegen\csharp\8.0.0\csharp-8.0.0.jar

Z:\git\javacc\javacc\javacc>set CP=C:\Users\FrancisANDRE\.m2\repository\org\javacc\parser\8.0.0\parser-8.0.0.jar;C:\Users\FrancisANDRE\.m2\repository\org\javacc\codegen\java\8.0.0\java-8.0.0.jar;C:\Users\FrancisANDRE\.m2\repository\org\javacc\codegen\cpp\8.0.0\cpp-8.0.0.jar

Z:\git\javacc\javacc\javacc>java -cp C:\Users\FrancisANDRE\.m2\repository\org\javacc\parser\8.0.0\parser-8.0.0.jar;C:\Users\FrancisANDRE\.m2\repository\org\javacc\codegen\java\8.0.0\java-8.0.0.jar;C:\Users\FrancisANDRE\.m2\repository\org\javacc\codegen\cpp\8.0.0\cpp-8.0.0.jar javacc -CODE_GENERATOR=Java -OUTPUT_DIRECTORY=tmp examples\java\Lookahead\src\main\javacc\Example1.jj
Java Compiler Compiler Version 8.0.0 (Parser Generator)
(type "javacc" with no arguments for help)
Reading from file examples\java\Lookahead\src\main\javacc\Example1.jj . . .
Warning: Output directory "tmp" does not exist. Creating the directory.
Warning: Choice conflict in [...] construct at line 60, column 7.
         Expansion nested within construct and expansion following construct
         have common prefixes, one of which is: "c"
         Consider using a lookahead of 2 or more for nested expansion.
File "Example.java" does not exist.  Will create one.
File "ExampleTokenManager.java" does not exist.  Will create one.
File "Token.java" does not exist.  Will create one.
File "TokenMgrError.java" does not exist.  Will create one.
File "ParseException.java" does not exist.  Will create one.
File "ExampleConstants.java" does not exist.  Will create one.
Warning: Token.java: File is obsolete.  Please rename or delete this file so that a new one can be generated for you.
Warning: Token.java file   version: 8.0.0 javacc version: 8.0
File "SimpleCharStream.java" does not exist.  Will create one.
Parser generated with 0 errors and 4 warnings.

Z:\git\javacc\javacc\javacc>endlocal

Z:\git\javacc\javacc\javacc>type tmp\Token.java
/* Generated By:JavaCC: Do not edit this line. Token.java Version 8.0 */
/* JavaCCOptions:TOKEN_EXTENDS=,KEEP_LINE_COLUMN=true,SUPPORT_CLASS_VISIBILITY_PUBLIC=true */
/**
 * Describes the input token stream.
 */

public class Token {
```

**Comment by mbrigl** [comment]
You are right, fixed with #43

**Comment by zosrothko** [comment]
Verified
